### PR TITLE
fix(api): send msUsername in get session user endpoint

### LINKED ISSUE #54049: Add msusername to get-session-user
See comment: https://github.com/freeCodeCamp/freeCodeCamp/pull/53934#issuecomment-1975230757

### PR DESCRIPTION
**Pull request by Nirajn2311**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #54049

### CODE REVIEW

**Review by moT01 — APPROVED** [review]
LGTM 🎉 

Only thing I potentially see is - do we want to delete the msUsername in the test after the test runs?

File: api/src/routes/user.test.ts [hunk 1]
```diff
@@ -582,6 +582,27 @@ describe('userRoutes', () => {
...
+
+        const msUsernames = await fastifyTestInstance.prisma.msUsername.count();
+        expect(msUsernames).toBe(1);
+
+        const response = await superGet('/user/get-session-user');
+
+        const { msUsername } = response.body.user.foobar;
+
+        expect(msUsername).toBe(msUsernameData.msUsername);
+      });
```
Comment (ShaunSHamilton) [inline api/src/routes/user.test.ts:604]:
```suggestion
      test('GET returns the msUsername if it exists', async () => {
        await fastifyTestInstance.prisma.msUsername.create({
          data: msUsernameData[0] as (typeof msUsernameData)[0]
        });

        const msUsernames = await fastifyTestInstance.prisma.msUsername.count();
        expect(msUsernames).toBe(1);

        const response = await superGet('/user/get-session-user');

        const { msUsername } = response.body.user.foobar;

        expect(msUsername).toBe(msUsernameData[0]?.msUsername);
      });
```

File: api/src/routes/user.ts [hunk 2]
```diff
@@ -515,11 +515,17 @@ export const userGetRoutes: FastifyPluginCallbackTypebox = (
           where: { userId: req.session.user.id }
         });
 
-        const [userToken, user, completedSurveys] = await Promise.all([
-          userTokenP,
-          userP,
-          completedSurveysP
-        ]);
+        const msUsernameP = fastify.prisma.msUsername.findFirst({
+          where: { userId: req.session.user.id }
```
Comment (ShaunSHamilton) [inline api/src/routes/user.ts:519]: After Oliver's PR, I believe this is what we now want:

```suggestion
          where: { userId: req.user?.id }
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Create many to create one is a bit odd.

**Review by ShaunSHamilton — APPROVED** [review]

**Comment by ShaunSHamilton** [comment]
Thanks, Niraj

**Pull request merged**